x/clp/types/params.go

```diff
@@ -33,66 +33,61 @@ func ParamKeyTable() paramtypes.KeyTable {
-		PmtpExternalWeight:     pmtpExternalWeight,
-		PmtpStartBlock:         pmtpStartBlock,
-		PmtpEndBlock:           pmtpEndBlock,
+		MinCreatePoolThreshold:   minThreshold,
+		PmtpPeriodGovernanceRate: pmtpGovernanaceRate,
+		PmtpPeriodEpochLength:    pmtpEpochLength,
+		PmtpPeriodStartBlock:     pmtpStartBlock,
+		PmtpPeriodEndBlock:       pmtpEndBlock,
 	}
 }
 
 // ParamSetPairs - Implements params.ParamSet
 func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
 	return paramtypes.ParamSetPairs{
 		paramtypes.NewParamSetPair(KeyMinCreatePoolThreshold, &p.MinCreatePoolThreshold, validateMinCreatePoolThreshold),
-		paramtypes.NewParamSetPair(KeyPmtpNativeWeight, &p.PmtpNativeWeight, validatePmtpNativeWeight),
-		paramtypes.NewParamSetPair(KeyPmtpExternalWeight, &p.PmtpExternalWeight, validatePmtpExternalWeight),
-		paramtypes.NewParamSetPair(KeyPmtpStartBlock, &p.PmtpStartBlock, validatePmtpStartBlock),
-		paramtypes.NewParamSetPair(KeyPmtpEndBlock, &p.PmtpEndBlock, validatePmtpEndBlock),
+		paramtypes.NewParamSetPair(KeyPmtpPeriodGovernanceRate, &p.PmtpPeriodGovernanceRate, validatePmtpPeriodGovernanceRate),
+		paramtypes.NewParamSetPair(KeyPmtpEpochLength, &p.PmtpPeriodEpochLength, validatePmtpPeriodEpochLength),
+		paramtypes.NewParamSetPair(KeyPmtpStartBlock, &p.PmtpPeriodStartBlock, validatePmtpStartBlock),
+		paramtypes.NewParamSetPair(KeyPmtpEndBlock, &p.PmtpPeriodEndBlock, validatePmtpEndBlock),
 	}
 }
 
 // DefaultParams defines the parameters for this module
 func DefaultParams() Params {
 	return Params{
-		MinCreatePoolThreshold: DefaultMinCreatePoolThreshold,
-		PmtpNativeWeight:       sdk.NewDecWithPrec(5, 1),
-		PmtpExternalWeight:     sdk.NewDecWithPrec(5, 1),
-		PmtpStartBlock:         DefaultPmtpStartBlock,
-		PmtpEndBlock:           DefaultPmtpEndBlock,
+		MinCreatePoolThreshold:   DefaultMinCreatePoolThreshold,
+		PmtpPeriodGovernanceRate: sdk.NewDecWithPrec(5, 1),
+		PmtpPeriodEpochLength:    7,
+		PmtpPeriodStartBlock:     DefaultPmtpStartBlock,
+		PmtpPeriodEndBlock:       DefaultPmtpEndBlock,
 	}
 }
 
 func (p Params) Validate() error { // TODO determine all checks
 	if err := validateMinCreatePoolThreshold(p.MinCreatePoolThreshold); err != nil {
 		return err
 	}
-	if err := validatePmtpNativeWeight(p.PmtpNativeWeight); err != nil {
+	if err := validatePmtpPeriodGovernanceRate(p.PmtpPeriodGovernanceRate); err != nil {
 		return err
 	}
-	if err := validatePmtpExternalWeight(p.PmtpExternalWeight); err != nil {
+	if err := validatePmtpPeriodEpochLength(p.PmtpPeriodEpochLength); err != nil {
 		return err
 	}
-	if err := validatePmtpStartBlock(p.PmtpStartBlock); err != nil {
+	if err := validatePmtpStartBlock(p.PmtpPeriodStartBlock); err != nil {
 		return err
 	}
-	if err := validatePmtpEndBlock(p.PmtpEndBlock); err != nil {
+	if err := validatePmtpEndBlock(p.PmtpPeriodEndBlock); err != nil {
 		return err
 	}
-	if p.PmtpEndBlock < p.PmtpStartBlock {
+	if p.PmtpPeriodEndBlock < p.PmtpPeriodStartBlock {
```

Comment:
@Jedi2002 Like you said in the doc, let's check that [(policy end date - policy start date ) % epoch ] is zero